sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -767,6 +729,85 @@ private void processMinidumpFiles() {
                 }
             }
         }
+
+        /* Remove the minidump subfolders from previous sessions. */
+        ErrorLogHelper.removeStaleMinidumpSubfolders();
+    }
+
+    /**
+     * Process the minidump, save an error log with a reference to the minidump, move the minidump file to the 'pending' folder.
+     *
+     * @param minidumpFile   a file where an ndk crash is saved.
+     * @param minidumpFolder a folder that contains device info and a minidump file.
+     */
+    private void processSingleMinidump(File minidumpFile, File minidumpFolder) {
+        /* Create missing files from the native crash that we detected. */
+        AppCenterLog.debug(LOG_TAG, "Process pending minidump file: " + minidumpFile);
+        long minidumpDate = minidumpFile.lastModified();
+        File dest = new File(ErrorLogHelper.getPendingMinidumpDirectory(), minidumpFile.getName());
+        Exception modelException = new Exception();
+        modelException.setType("minidump");
+        modelException.setWrapperSdkName(WRAPPER_SDK_NAME_NDK);
+        modelException.setMinidumpFilePath(dest.getPath());
+        ManagedErrorLog errorLog = new ManagedErrorLog();
+        errorLog.setException(modelException);
+        errorLog.setTimestamp(new Date(minidumpDate));
+        errorLog.setFatal(true);
+        errorLog.setId(ErrorLogHelper.parseLogFolderUuid(minidumpFolder));
+
+        /* Lookup app launch timestamp in session history. */
+        SessionContext.SessionInfo session = SessionContext.getInstance().getSessionAt(minidumpDate);
+        if (session != null && session.getAppLaunchTimestamp() <= minidumpDate) {
```

Comment:
Check here if `minidumpDate` is not **0**. `lastModified()` can return **0L** if the file does not exist or if an I/O error occurs (the latter can happen).

Reply:
I think in this case `SessionContext.getInstance().getSessionAt(minidumpDate)` will return `null` and anyway we'll get to the else clause. We receive `mAppLaunchTimestamp` for the session through `System.currentTimeMillis()` and there is no valid chance it would be 0.